### PR TITLE
[OFFLINE] Download UI for downloading courses

### PR DESCRIPTION
**Pull request by ITPini**
## Description

This increment implements a UI element for course cards, designed specifically for downloading courses. It's important to note that this pull request focuses solely on the UI aspect; the actual download functionality is not yet implemented. The design has received approval from the Product Owner.

## Changes

- Added a download UI for course cards
- Added download animation asset
- Removed outdated unit test
- Dependency added: `lottie-react-native` version 5.1.3.

## Related Issues

None.

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [ ] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [ ] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 
Branch: `dev`

## Notes for Reviewers

Please pay special attention to the new download UI integration and its impact on overall performance. This has not been tested on older Android phones, but shouldn't cause any issues.

### CODE REVIEW

**Review by JackSMunn — APPROVED** [review]
Approved, did not test locally though

**Review by PrometheusBork — APPROVED** [review]
Works fine for me! Good work!

File: eml/assets/animations/downloadAnimation.json
Comment (sofiiagran) [inline eml/assets/animations/downloadAnimation.json:1]: Is this supposed to be like this?

**Review by sofiiagran — APPROVED** [review]
Looks good to me :) Good job guys

File: eml/assets/animations/downloadAnimation.json
Comment (ITPini) [reply eml/assets/animations/downloadAnimation.json:1]: Yes, it contains the whole animation but is not very human-readable :(

**Pull request merged**